core/src/main/java/feast/core/model/Source.java

```diff
@@ -16,43 +16,35 @@
  */
 package feast.core.model;
 
-import com.google.common.collect.Sets;
-import com.google.protobuf.Message;
+import com.google.protobuf.TextFormat;
 import feast.core.SourceProto;
 import feast.core.SourceProto.KafkaSourceConfig;
 import feast.core.SourceProto.Source.Builder;
 import feast.core.SourceProto.SourceType;
 import io.grpc.Status;
 import java.util.Objects;
-import java.util.Set;
-import javax.persistence.Column;
+import javax.persistence.*;
 import javax.persistence.Entity;
-import javax.persistence.Id;
-import javax.persistence.Table;
+import lombok.Getter;
 import lombok.Setter;
 
+@Getter
 @Setter
 @Entity
 @Table(name = "sources")
 public class Source {
 
-  private static final Set<String> KAFKA_OPTIONS = Sets.newHashSet("bootstrapServers");
+  /** Source Id */
+  @Id @GeneratedValue private long id;
```

Comment:
Doesn't this mean that a new record is created for every source pushed to the db, even if there was a previously applied fs sharing the same source?

Reply:
Yes it does. I thought about unifying these, but I think it would be unintuitive for one feature set to be able to modify the source of another feature set. Ideally a source would be configured by the administrator and the user should just select it, at which point sharing sources would make more sense in my opinion.

Reply:
It's not possible for a feature set to modify the source of another feature set, if a feature set's source is updated a new entry will be written to the db, distinct from the old one (still in use by other feature sets).

Reply:
Ok. Didn't realize that. So it acts pretty similarly to how it's written here with some deduplication?

Should I change anything or are you happy with this implementation as a generalization step?

Reply:
No, this implementation won't work once you're able to alter feature set sources, unfortunately. It will break job creation.